spring-integration-smb/src/main/java/org/springframework/integration/smb/outbound/SmbMessageHandler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.smb.outbound;
+
+import org.springframework.integration.file.remote.handler.FileTransferringMessageHandler;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.smb.session.SmbRemoteFileTemplate;
+import org.springframework.integration.smb.session.SmbSessionFactory;
+
+import jcifs.smb.SmbFile;
+
+/**
+* The SMB specific {@link FileTransferringMessageHandler} extension. Based on the
+* {@link SmbRemoteFileTemplate}.
+*
+* @author Gregory Bragg
+*
+* @since 1.2.2
+*
+* @see SmbRemoteFileTemplate
+*/
+public class SmbMessageHandler extends FileTransferringMessageHandler<SmbFile> {
+
+	@SuppressWarnings("deprecation")
+	public SmbMessageHandler(SessionFactory<SmbFile> sessionFactory) {
```

Comment:
I think in addition to this one, I'd like to see this ctor in addition:
```
public SmbMessageHandler(SessionFactory<SmbFile> sessionFactory, FileExistsMode mode) {

```

and do the respective logic depending on the `FileExistsMode`

Reply:
hoe about this:
	public SmbMessageHandler(SessionFactory<SmbFile> sessionFactory, FileExistsMode mode) {
		this(sessionFactory);
	}

Reply:
Why? Everything is OK. Append, replace, delete etc.
I mean right now we expose only one ctor with the default mode.
To avoid `SmbRemoteFileTemplate` pollution I just suggest to have it based on the `SessionFactory`. Delegate to respective super and call `setReplaceFile()` on the factory when it makes sense based on the provided mode.

Reply:
Ok I'm obviously misunderstanding here... my apologies... this is a new constructor that doesn't follow any of the base constructors in FileTransferringMessageHandler?

What am I missing?

Reply:
Yes, a new one.
Well, it indeed does not, but it is going to delegate to super based on the built `SmbRemoteFileTemplate`.

Makes sense?

Reply:
Ok well let me try... not sure I understand the use case fully.